doc/getting_started/index.rst

```diff
@@ -3,319 +3,372 @@
+
+      .. _Homebrew site: https://brew.sh/
+
+   .. group-tab:: Windows
+
+      .. note:: If you're running Windows 10 (version 1809 or later)
+         you can run Ubuntu binaries natively within the built-in Windows
+         Subsystem for Linux (WSL). Using WSL will also let you to use
+         the Zephyr SDK. `Install the Windows Subsystem for Linux (WSL)
+         <https://msdn.microsoft.com/en-us/commandline/wsl/install_guide>`__, and
+         then follow the Linux (Ubuntu) Getting Started instructions.
+
+      These instructions assume you are using the Windows ``cmd.exe``
+      command prompt. Some of the details, such as setting environment
+      variables, may differ if you are using PowerShell.
+
+      An easy way to install native Windows dependencies is to first install
+      `Chocolatey`_, a package manager for Windows.  If you prefer to install
+      dependencies manually, you can also download the required programs from their
+      respective websites and verify they can be found on your PATH.
+
+      #. Install :program:`Chocolatey` by following the instructions on the
+         `Chocolatey install`_ page.
+
+      #. Open a command prompt (``cmd.exe``) as an **Administrator** (press the
+         Windows key, type "cmd.exe" in the prompt, then right-click the result and
+         choose "Run as Administrator").
+
+      #. Disable global confirmation to avoid having to confirm
+         installation of individual programs:
+
+         .. code-block:: console
+
+            choco feature enable -n allowGlobalConfirmation
+
+      #. Install CMake:
 
-Clone the Zephyr Repositories
-=============================
+         .. code-block:: console
 
-Clone all of Zephyr's repositories in a new :file:`zephyrproject` directory:
+            choco install cmake --installargs 'ADD_CMAKE_TO_PATH=System'
 
-.. code-block:: console
+      #. Install the rest of the tools:
 
-   west init zephyrproject
-   cd zephyrproject
-   west update
+         .. code-block:: console
 
-You can replace :file:`zephyrproject` with another directory name. West creates
-the directory if it doesn't exist. See :ref:`west-multi-repo` for more details.
+            choco install git python ninja dtc-msys2 gperf
 
-.. important::
+      #. Close the Administrator command prompt window.
 
-   You need to run ``west update`` any time :file:`zephyr/west.yml` changes.
-   This command keeps :ref:`modules` in the :file:`zephyrproject` folder in sync
-   with the code in the zephyr repository, so they work correctly together.
+      Install needed python packages with these commands:
 
-   Some examples when ``west update`` is needed are: whenever you
-   pull the :file:`zephyr` repository, switch branches in it, or perform a ``git
-   bisect`` inside of it.
+      .. code-block:: bash
 
-.. warning::
+         pip3 install -r https://raw.githubusercontent.com/zephyrproject-rtos/zephyr/master/scripts/requirements.txt
```

Comment:
That's a nice solution for getting the requirements.txt locally for pip3 to use, but it does mess up the "Install tools" then "get source" flow (I think for the better because it eliminates the hard version reference).

Reply:
>  (I think for the better because it eliminates the hard version reference).

what version reference? are you referring to 'master'? This is just the same, it is just not implied.

In general I think we can solve this problem by introducing macros for releases and branches and avoid hardcoding the branch or the version so that such documentation has the right reference whether we are on master or on a release.

Reply:
> But I think that in general, content like this, as well as directives like `:zephyr_file:` suffer from the problem that they do not produce stable references to artifacts that are relevant to the content they are meant to document.

zephyr_file actually should work on branches, at least the implementation allows versioned references, maybe we are not using it correctly.

Reply:
Can’t use a reST substitution in the code block because that content isn’t processed. Sigh.

Reply:
> Can’t use a reST substitution in the code block because that content isn’t processed. Sigh.

You can use a parsed-literal if you're willing to give up syntax highlighting. Otherwise your only option is to write a Python extension that uses a variable set in conf.py to generate the code block nodes directly into the docutils DOM. But that's of course more complex.

Reply:
> what version reference? are you referring to 'master'? This is just the same, it is just not implied.

Right, it always refers to master, which becomes instantly wrong the moment the release is cut and the HTML stays the same while master changes.

> In general I think we can solve this problem by introducing macros for releases and branches and avoid hardcoding the branch or the version so that such documentation has the right reference whether we are on master or on a release.

Yes, but that has the problems @dbkinder pointed out. I've dealt with the same issue in the foundries documentation and the parsed-literal or extension options are the only ones I could find.

Reply:
I agree that pointing to the upstream `requirements.txt` is not going to work with documentation for previous releases, so it's better to always rely on the actual file on the disk post `west init`, because otherwise this will fail:

```
pip3 install -r https://raw.githubusercontent.com/zephyrproject-rtos/zephyr/master/scripts/requirements.txt
west init --mr v1.14.0
```
since the requirements of master will have been installed to clone v1.14.0.

Reply:
> west init --mr v1.14.0

That works for west init, but I just want to reiterate that this is a much more general concern about an issue throughout the documentation.